Content.Server/Zombies/ZombieSystem.cs

```diff
@@ -69,46 +89,67 @@ public override void Update(float frameTime)
             var query = EntityQueryEnumerator<PendingZombieComponent, DamageableComponent, MobStateComponent>();
             var curTime = _timing.CurTime;
 
-            var zombQuery = EntityQueryEnumerator<ZombieComponent, DamageableComponent, MobStateComponent>();
-
             // Hurt the living infected
-            while (query.MoveNext(out var uid, out var comp, out var damage, out var mobState))
+            while (query.MoveNext(out var uid, out var pending, out var damage, out var mobState))
             {
                 // Process only once per second
-                if (comp.NextTick + TimeSpan.FromSeconds(1) > curTime)
+                if (pending.NextTick + TimeSpan.FromSeconds(1) > curTime)
                     continue;
 
-                comp.NextTick = curTime;
+                pending.NextTick = curTime;
 
-                comp.InfectedSecs += 1;
+                if (pending.FirstTurnAllowed != TimeSpan.Zero)
+                    continue;
+
+                pending.InfectedSecs += 1;
                 // See if there should be a warning popup for the player.
-                if (comp.InfectionWarnings.TryGetValue(comp.InfectedSecs, out var popupStr))
+                if (pending.Settings.InfectionWarnings.TryGetValue(pending.InfectedSecs, out var popupStr))
```

Comment:
See my issue for why these changes need fixing.

Reply:
I've simplified all this.